### ISSUE TITLE
How to rotate a textri() sprite horizontally/vertically

### ISSUE THREAD

**Issue by IsmaelSaad** [nesbox/TIC-80#2169]
I've finally understood the txtri function and a function called rspr to rotate sprites but now I want to rotate my sprite horizontally/vertically, how could I do that?

Thanks in advance.

**Comment by bztsrc** [comment]
> I want to rotate my sprite horizontally/vertically

You can't really do that. With the [spr](https://github.com/nesbox/TIC-80/wiki/spr) either you rotate the sprite by 0, 90, 180 or 270 degrees (using the `rotate` parameter), or you flip it horizontally/vertically (with the `flip` parameter). Flipped vertically and rotated by 180 will both provide the sprite upside down, but the results will be each other's mirrored versions.
```
012    rotate 180     876
345    ---------->    543
678                   210

012    flip vert      678
345    ---------->    345
678                   012
```

As for [ttri](https://github.com/nesbox/TIC-80/wiki/ttri), just swap the U or V parameters for the edges. For example to flip horizontally, the original u1=0, u2=2 becomes u1=2, u2=0:
```
(u1,v1) (u2,v1)          (u2,v1) (u1,v1)
      012                      012
      345        ----->        345
      678                      678
(u1,v2) (u2,v2)          (u2,v2) (u1,v2)
```

Cheers,
bzt

**Comment by IsmaelSaad** [comment]
Hi again,
This is the rspr() function:
``` lua
function rspr(sx,sy,scale,angle,mx,my,mw,mh,key,useMap)

  --  this is fixed , to make a textured quad
  --  X , Y , U , V
  local sv ={{-1,-1, 0,0},
             { 1,-1, 0.999,0},
             {-1,1,  0,0.999},
             { 1,1,  0.999,0.999}}

	local rp = {} --  rotated points storage

  --  the scale is mw ( map width ) * 4 * scale 
  --  mapwidth is * 4 because we actually want HALF width to center the image
  local scalex = (mw<<2) * scale
  local scaley = (mh<<2) * scale
  --  rotate the quad points
  for p=1,#sv do 
    -- apply scale
    local _sx = sv[p][1] * scalex 
		local _sy = sv[p][2] * scaley
    -- apply rotation
		local a = -angle
		local rx = _sx * math.cos(a) - _sy * math.sin(a)
		local ry = _sx * math.sin(a) + _sy * math.cos(a)
    -- apply transform
    sv[p][1] = rx + sx
    sv[p][2] = ry + sy
    -- scale UV's 
    sv[p][3] = (mx<<3) + (sv[p][3] * (mw<<3))
    sv[p][4] = (my<<3) + (sv[p][4] * (mh<<3))
  end
  -- draw two triangles for the quad
  textri( sv[1][1],sv[1][2], 
          sv[2][1],sv[2][2],  
          sv[3][1],sv[3][2], 
          sv[1][3],sv[1][4], --u1 v1
          sv[2][3],sv[2][4], --u2 v2
          sv[3][3],sv[3][4], 
          useMap,key)

  textri( sv[2][1],sv[2][2],
          sv[3][1],sv[3][2],
          sv[4][1],sv[4][2],
          sv[2][3],sv[2][4],--u1 v2
          sv[3][3],sv[3][4],--u2 v2
          sv[4][3],sv[4][4],
          useMap,key)
end
```
I just changed the u1-u2, v1-v2 variables for each txtri function.
``` lua
  textri( sv[1][1],sv[1][2], 
          sv[2][1],sv[2][2],  
          sv[3][1],sv[3][2], 
          sv[1][4],sv[1][3], --u1 v1
          sv[2][4],sv[2][3], --u2 v2
          sv[3][3],sv[3][4], 
          useMap,key)
  textri( sv[2][1],sv[2][2],
          sv[3][1],sv[3][2],
          sv[4][1],sv[4][2],
          sv[2][4],sv[2][3],--u1 v2
          sv[3][4],sv[3][3],--u2 v2
          sv[4][3],sv[4][4],
          useMap,key)

```At first I thought it would work but I tried and it doesn't work. Am I doing something wrong?
Thanks in advance.

**Comment by bztsrc** [comment]
Not sure what `textri` supposed to be, the function for displaying a textured triangle is called [ttri](https://github.com/nesbox/TIC-80/wiki/ttri) and that latter does work for me.
![ttri](https://user-images.githubusercontent.com/12779069/233224633-9c805432-0f83-48aa-ad67-b6a3f82a11c1.png)

Hope this helps,
bzt

**Comment by scambier** [comment]
`textri` is the older deprecated version of `ttri`: https://github.com/nesbox/TIC-80/wiki/textri

**Comment by bztsrc** [comment]
> textri is the older deprecated version of ttri

Ah I see. So that's why the syntax highlighter doesn't recognize it and why it isn't listed on the wiki.

Anyway, swapping the U,V coordinates does work with `textri` too...
![textri](https://user-images.githubusercontent.com/12779069/233830939-4c91d593-3a72-4147-ad5e-114643f03d6a.png)

Cheers,
bzt

**Comment by IsmaelSaad** [comment]
Hi, I've found the solution to the problem. Basically, what bzt said about changing the UV coordinates in the ttri() function does indeed work, but the thing is that in my case we have 2 triangles that draw the desired sprite. I think that with this image I'll explain myself better:
![flipttriexplanation](https://user-images.githubusercontent.com/107066576/233840621-b682725b-3612-491b-a3d5-3f652470a28f.png)
What I'm trying to explain with this image is that if we interchange the UV variables with heach other we could manage to draw the sprite but flipped horizontally. I assume that works the same with vertical flips.
In addition, I added a little condition in rspr() following this method. ("flip" is a variable I added to flip the sprite when I need to)
``` lua
if flip==1 then
	sv[1][1],sv[3][1]=sv[3][1],sv[1][1]
	sv[1][2],sv[3][2]=sv[3][2],sv[1][2]
	sv[2][1],sv[4][1]=sv[4][1],sv[2][1]
	sv[2][2],sv[4][2]=sv[4][2],sv[2][2]
end
```
Best regards,
Izma
